Prog/ACS_2012_NC.sas

```diff
@@ -0,0 +1,750 @@
+  EMPSTATD    EMPSTATD_f.
+  LABFORCE    LABFORCE_f.
+  OCC1950     OCC1950_f.
+  OCC1990     OCC1990_f.
+  OCC2010     OCC2010_f.
+  IND1950     IND1950_f.
+  IND1990     IND1990_f.
+  CLASSWKR    CLASSWKR_f.
+  CLASSWKRD   CLASSWKRD_f.
+  WKSWORK2    WKSWORK2_f.
+  UHRSWORK    UHRSWORK_f.
+  WRKLSTWK    WRKLSTWK_f.
+  ABSENT      ABSENT_f.
+  LOOKING     LOOKING_f.
+  AVAILBLE    AVAILBLE_f.
+  WRKRECAL    WRKRECAL_f.
+  WORKEDYR    WORKEDYR_f.
+  MIGRATE1    MIGRATE1_f.
+  MIGRATE1D   MIGRATE1D_f.
+  MIGPLAC1    MIGPLAC1_f.
+  VETDISAB    VETDISAB_f.
+  DIFFREM     DIFFREM_f.
+  DIFFPHYS    DIFFPHYS_f.
+  DIFFMOB     DIFFMOB_f.
+  DIFFCARE    DIFFCARE_f.
+  DIFFSENS    DIFFSENS_f.
+  DIFFEYE     DIFFEYE_f.
+  DIFFHEAR    DIFFHEAR_f.
+  VETSTAT     VETSTAT_f.
+  VETSTATD    VETSTATD_f.
+  VET01LTR    VET01LTR_f.
+  VET90X01    VET90X01_f.
+  VET75X90    VET75X90_f.
+  VETVIETN    VETVIETN_f.
+  VET55X64    VET55X64_f.
+  VETKOREA    VETKOREA_f.
+  VET47X50    VET47X50_f.
+  VETWWII     VETWWII_f.
+  VETOTHER    VETOTHER_f.
+  PWSTATE2    PWSTATE2_f.
+  TRANWORK    TRANWORK_f.
+  CARPOOL     CARPOOL_f.
+  RIDERS      RIDERS_f.
+  GCHOUSE     GCHOUSE_f.
+  GCMONTHS    GCMONTHS_f.
+  GCRESPON    GCRESPON_f.
+;
+
+  *** UI-created format for race Yes/No vars. ***;
+  format racamind racasian racblk racother racpacis racwht uracyn.;
+
+  ** Create unique PUMA ID **;
+
+  %upuma() 
+  
+  ** Recode n/a values **;
+  
+  array inca{*}  incwage incbus00 incretir incinvst  ;
+
+  do i = 1 to dim( inca );
+    if inca{i} = 999999 then inca{i} = .n;
+  end;
+
+  array incb{*} incss incwelfr incsupp incother owncost;
+
+  do i = 1 to dim( incb );
+    if incb{i} = 99999 then incb{i} = .n;
+  end;
+
+  array incc{*} ftotinc hhincome inctot valueh;
+
+  do i = 1 to dim( incc );
+    if incc{i} = 9999999 then incc{i} = .n;
+  end;
+
+  ** Utility cost codes 9993-9998 indicate no charge, not used, or costs included in rent/condo fee **;
+
+  array incd{*} costelec costgas costfuel costwatr;
+  
+  do i = 1 to dim( incd );
+    if incd{i} >= 9992 then incd{i} = .n;
+  end;
+  
+  ** 2-digit N/A codes **;
+  
+  array ince{*} eldch yngch;
+  
+  do i = 1 to dim( ince );
+    if ince{i} = 99 then ince{i} = .n;
+  end;
+  
+  ** yrnatur **;
+  
+  array incf{*} yrnatur;
+  
+  do i = 1 to dim( incf );
+    if incf{i} >= 9997 then incf{i} = .n;
+  end;
+  
+  ** Earnings, education, and occupational status scores **;
+  
+  array incg{*} erscor50 erscor90 edscor50 edscor90 npboss50 npboss90;
+  
+  do i = 1 to dim( incg );
+    if incg{i} = 999.9 then incg{i} = .n;
+  end;
+
+  /*
+  ** HUD income categories, will be calculated in the tabulation program **;
+  
+  %hud_inc_all_NC()
+*/
+
+  ** Output files by state **;
+  
+  select ( statefip );
+    when( &FIPS_NC ) output ACS_&yr._NC;
+
+    otherwise do;
+      %err_put( msg="Invalid state code: " _n_= serial= statefip= pernum= )
+    end;
+  end;
+  
+  drop i;
+
+run;
+
+proc contents data= ACS_2017_NC;
```

Comment:
remove this proc contents (for2017 - causing error and one runs with the finalize macro). - i think this error is on the other programs, please correct.